### ISSUE TITLE
assorted new 2.12.x failures

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#526]
https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1418/consoleFull

```
[info] >>> The dbuild result is------------: FAILED (failed:
  scalaprops, scalameta, akka-sse, algebra, play-ws, atto)
```

same in 1417, so all probably not intermittent

 - [X] scalaprops, algebra, play-ws, atto: disable `-Xfatal-warnings`
 - [X] scalameta: needs https://github.com/scala/scala/pull/5844
 - [X] scalamock: regression in specs2 (https://github.com/etorreborre/specs2/commit/e4e9ab7)
 - [X] akka-sse:

```
[akka-sse] [warn] an unexpected type representation reached the compiler backend while compiling package.scala: <error>. If possible, please file a bug on issues.scala-lang.org.
[akka-sse:error] scala.MatchError: <error> (of class scala.reflect.internal.Types$ErrorType$)
[akka-sse:error] 	at scala.tools.nsc.backend.jvm.BTypesFromSymbols.typeToBType(BTypesFromSymbols.scala:212)
[akka-sse:error] 	at scala.tools.nsc.backend.jvm.BTypesFromSymbols.methodBTypeFromMethodType(BTypesFromSymbols.scala:146)
```

**Comment by SethTisue** [comment]
akka-sse: no recent commits in the upstream repo. last green run was with 2.12.2-bin-1ef37f0. nontrivial Scala PRs merged since then: 5402 (unused warnings), scala/scala#5816 (apply/unapply)

@adriaanm and @lrytz both suspect 5816

**Comment by SethTisue** [comment]
warnings fix in 6430c0a822582aaf0089c36a6c1697dbf779d460, https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1419/consoleFull in progress

**Comment by SethTisue** [comment]
now scalaprops needs the same JSExportDescendentObjects fix utest got

UPDATE: f61b173b3b07759960a4057a30853f76093521af, newer test run https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1420/consoleFull

**Comment by SethTisue** [comment]
refreshed scalameta fork in scalacommunitybuild org, so we're testing the latest there. dbuild tweaks in c420055427d537b76161b473d3ebf32271cb2ed3. new test run https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1421/consoleFull

**Comment by SethTisue** [comment]
5844 merged, new release-main job: https://scala-ci.typesafe.com/job/scala-2.12.x-release-main/756/consoleFull

**Comment by SethTisue** [comment]
akka-sse thing is now scala/bug#10261

**Comment by SethTisue** [comment]
as of run 1421, just 3 failures: scalameta, ​scalamock, ​akka-sse. run 1422 should fix scalameta; https://github.com/scala/scala/pull/5845 isn't merged yet but should fix akka-sse; hopefully the scalamock thing is a transient, timeout-type thing (it looks like it from the name of the failed test)

**Comment by SethTisue** [comment]
the scalamock thing is not transient; `org.scalamock.test.specs2.SuiteScopePresetMockParallelTest` fails in every run

**Comment by adriaanm** [comment]
> the scalamock thing is not transient; org.scalamock.test.specs2.SuiteScopePresetMockParallelTest fails in every run

investigating after lunch

**Comment by SethTisue** [comment]
scalameta fixed by #527

**Comment by SethTisue** [comment]
in local test: `[info] Project akka-sse----------------: SUCCESS (project rebuilt ok)`

**Comment by SethTisue** [comment]
scalamock thing was addressed by freezing specs2 in 65c3a820f15aa702c66052d30cf76cbb70a33c6c and b433526d22cc7fced09f400f59c6f3a22807f7ef
